### ISSUE TITLE
Allow creation of notifications with buttons

### ISSUE THREAD

**Issue by tommoor** [atom/atom-shell#1399]
It would be awesome to be able to create notifications with action buttons like so: 

![image](https://cloud.githubusercontent.com/assets/380914/7094023/e31445a8-df72-11e4-8d54-3b669ce9a81b.png)

I believe from an API point of view this would be a simple addition and we could easily follow what's been laid out for chrome extension notifications here:

https://developer.chrome.com/apps/notifications#event-onButtonClicked

```javascript
var notification = new Notification('Mentioned by @imsh', {
  body: "Dude I really don't get it...",
  buttons: [{
    title: "Close"
  }, {
    title: "Show"
  }]
});
```

In summary, you would add a 'buttons' array into new Notification options and receive a onButtonClicked callback with the index of the button that was clicked.

Thoughts? I'm not upto date on whether Windows notifications allow for similar functionality...

**Comment by paulcbetts** [comment]
Only OS X allows you to have actions on notifications, both Windows and Linux do not :(

**Comment by tommoor** [comment]
On the brightside, that would make the implementation easier?! :laughing:

**Comment by zcbenz** [comment]
There is plenty of node modules that can send native notifications on OS X, though I'm not sure whether there exists one that can set actions. And you can always write one yourself.

The HTML5 notification API doesn't have concept of actions, and introducing a new set of APIs for a simple feature doesn't seem worthy to me.

So I'm closing this as won't fix.

**Comment by tommoor** [comment]
Understand the need for focus but disappointed none the less - this would be a great addition. Most of the time when you trigger a notification it has some sort of associated action.

**Comment by Sidnicious** [comment]
This would definitely be useful — "real" desktop apps on OS X can show notifications with actions, so it makes sense that Electron would expose this too.

**Comment by screenmeet** [comment]
This would be a great feature...

**Comment by dmblack** [comment]
Should this be closed? I was under the impression that libnotify does support actions? This doesn't cover Windows.. But if the Linux and macOS API's allow it..?

**Comment by SD-Gaming** [comment]
So far, that's still the case. sad

**Comment by shivam-bit** [comment]
Someone reading this and looking for something similar try [Node-Notifier](https://www.npmjs.com/package/node-notifier).
